regparser/commands/outline_depths.py

```diff
@@ -0,0 +1,33 @@
+import logging
+from regparser.tree.depth.derive import derive_depths
+
+import click
+
+logger = logging.getLogger(__name__)
+
+
+@click.command()
+@click.argument('markers', type=click.STRING, required=True)
+def outline_depths(markers) -> None:
+    """
+    Infer an outline's structure.
+    Return a list of outline depths for a given list of space-separated markers.
+    """
+
+    # Input is space-separated.
+    marker_list = markers.split(' ')
+    all_solutions = derive_depths(marker_list, [])
+    depths = {tuple(str(a.depth) for a in s) for s in all_solutions}.pop()
+
+    # Expected output is space-separated.
+    formatted_output = ' '.join(depths)
```

Comment:
Heads up that `Solution`s have a [`pretty_str`](https://github.com/eregs/regulations-parser/blob/da0a3d4cdef35c457ebc5b3c17221537b63714bc/regparser/tree/depth/derive.py#L38) method, which can be easier for humans to read. Might make sense as a parameter.